src/core/obj.js

```diff
@@ -252,6 +254,114 @@ var Catalog = (function CatalogClosure() {
       }
       return dest;
     },
+
+    get pageLabels() {
+      var obj = null;
+      try {
+        obj = this.readPageLabels();
+      } catch (ex) {
+        if (ex instanceof MissingDataException) {
+          throw ex;
+        }
+        warn('Unable to read page labels.');
+      }
+      return shadow(this, 'pageLabels', obj);
+    },
+    readPageLabels: function Catalog_readPageLabels() {
+      var xref = this.xref;
+      var pageLabels = [];
+      var obj = this.catDict.getRaw('PageLabels');
+      if (obj) {
+        var style = null;
+        var prefix = '';
+        var start = 1;
+
+        var numberTree = new NumberTree(obj, xref);
+        var nums = numberTree.getAll();
+        var currentLabel = '', currentIndex = 1;
+
+        for (var i = 0, ii = this.numPages; i < ii; i++) {
+          if (nums.hasOwnProperty(i)) {
+            var labelDict = nums[i];
+            assert(isDict(labelDict), 'The PageLabel is not a dictionary.');
+
+            var type = labelDict.get('Type');
+            assert(!type || (isName(type) && type.name === 'PageLabel'),
+                   'Invalid type in PageLabel dictionary.');
+
+            var s = labelDict.get('S') || Name.get('');
+            assert(isName(s), 'Invalid style in PageLabel dictionary.');
+            style = s.name;
+
+            prefix = labelDict.get('P') || '';
+            assert(isString(prefix), 'Invalid prefix in PageLabel dictionary.');
+
+            start = labelDict.get('St') || 1;
+            assert(isInt(start), 'Invalid start in PageLabel dictionary.');
+            currentIndex = start;
+          }
+
+          switch (style) {
+            case 'D':
+              currentLabel = '' + currentIndex;
+              break;
+            case 'R':
+            case 'r':
+              var ROMAN_NUMBERS = [
+                '', 'C', 'CC', 'CCC', 'CD', 'D', 'DC', 'DCC', 'DCCC', 'CM',
+                '', 'X', 'XX', 'XXX', 'XL', 'L', 'LX', 'LXX', 'LXXX', 'XC',
+                '', 'I', 'II', 'III', 'IV', 'V', 'VI', 'VII', 'VIII', 'IX'
+              ];
+              var number = currentIndex, pos;
+              currentLabel = '';
+
+              while (number >= 1000) {
+                currentLabel += 'M';
+                number -= 1000;
+              }
+              // Hundreds
+              pos = (number / 100) | 0;
+              number %= 100;
+              currentLabel += ROMAN_NUMBERS[pos];
+              // Tens
+              pos = (number / 10) | 0;
+              number %= 10;
+              currentLabel += ROMAN_NUMBERS[10 + pos];
+              // Ones
+              currentLabel += ROMAN_NUMBERS[20 + number];
+              break;
+            case 'A':
+            case 'a':
+              var LIMIT = 26; // Use only the characters A--Z, or a--z.
+              var letterIndex = currentIndex - 1;
+
+              var character = String.fromCharCode((letterIndex % LIMIT) + 0x41);
```

Comment:
`0x41` seems a bit magical here. Could you replace this with either a constant or a character (as I assume it is meant to indicate ASCII character `A`)?